### DISCUSSION TITLE
/tmp/pytest-of-$USER taking up a lot of space

### DISCUSSION THREAD

**Discussion by mimre25** [pytest-dev/pytest#12283] [Q&A]
Hi,

I just noticed that the pytest tmp-files take up a lot of space:
```console
~ du -sh /tmp/pytest-of-martin 
18G	/tmp/pytest-of-martin 
```

I searched through the issues a bit but I couldn't find anything that points at to such a huge usage.

Any idea what could cause that or how I could investigate this further?

For context: I'm using `micromamba` to separate envs and have multiple envs with pytest installed inside of them. 
I'm also using `tox` and work on many different projects - but 18GB seems just extremely excessive to me.

**Comment by RonnyPfannschmidt** [comment]
The seen size is indeed excessive and uncommon 

How many folders are directly inside that one?

**Reply by mimre25** [reply]
Only the 7 shown below and a symlink `pytest-current` which points at `pytest-9`
```console
➜  pytest-of-martin pwd
/tmp/pytest-of-martin                                       [0,02s]
➜  pytest-of-martin du -h --max-depth=1                   
2,0G	./pytest-5
101M	./pytest-9
5,3G	./pytest-8
2,4G	./pytest-6
1,4G	./pytest-7
2,0G	./pytest-4
4,4G	./pytest-2
```

**Reply by RonnyPfannschmidt** [reply]
I recommend exploring the retention policies that drop passing test tmpdirs earlier as it looks like your baseline is higher

**Comment by The-Compiler** [comment]
pytest itself doesn't store anything there, but your tests probably do. You might want to use something like [ncdu](https://dev.yorhel.nl/ncdu) or [gdu](https://github.com/dundee/gdu) (or graphical eqivalents) to figure out what's in there and where it's coming from.

I also find it odd that you have 7 such folders, though. [By default](https://docs.pytest.org/en/latest/reference/reference.html#confval-tmp_path_retention_count), pytest stores 3 folders, so I'm assuming you set `tmp_path_retention_count = 7` somewhere?

**Reply by mimre25** [reply]
I've ran a recursive grep over all my projects but couldn't find the settings anywhere.

Is there any other way to configure it? Maybe via an environment variable?
Do you know of any other tools affecting it? (Eg `tox` or `virtualenv`?)

Most space in there is used by folders like
```console
457M	/tmp/pytest-of-martin/pytest-8/popen-gw3/test_pyproject_deps_from_stati9/p/.tox
```

It turns out that `test_pyproject_deps_from_static` is a test in `tox` ([src](https://github.com/tox-dev/tox/blob/main/tests/tox_env/python/virtual_env/package/test_package_pyproject.py?plain=1#L158))

**Reply by RonnyPfannschmidt** [reply]
I believe enabling eagerly removing passed test folders will cut the size

**Reply by mimre25** [reply]
Since I don't necessarily have a say about such settings in bigger open source projects, I think my hands are tied.

As for now, I'll just delete it. 
And if it grows again, I'll take a shot a reproducing it. :slightly_smiling_face: 

Thanks, for all the input!

**Reply by RonnyPfannschmidt** [reply]
A start would be a upstream report 

The features that help are most likely more recent than the test support these projects put in place, it may turn into a welcome addition

**Reply by mimre25** [reply]
FYI: I've reported it here: https://github.com/tox-dev/tox/issues/3278
